lib/src/client.dart

```diff
@@ -521,13 +536,40 @@ class Client extends MatrixApi {
             loginTypes.map((f) => f.type ?? '').toSet(), supportedLoginTypes);
       }
 
-      return (wellKnown, versions, loginTypes);
+      return (versions, loginTypes);
     } catch (_) {
       homeserver = null;
       rethrow;
     }
   }
 
+  /// Gets discovery information about the domain. The file may include
+  /// additional keys, which MUST follow the Java package naming convention,
+  /// e.g. `com.example.myapp.property`. This ensures property names are
+  /// suitably namespaced for each application and reduces the risk of
+  /// clashes.
+  ///
+  /// Note that this endpoint is not necessarily handled by the homeserver,
+  /// but by another webserver, to be used for discovering the homeserver URL.
+  ///
+  /// In case is provided, [hostname] will be used in place of [homeserver] for
+  /// the HTTPS lookup. This is useful since the matrix homeserver does not
+  /// necessarily also host a copy of the `.well-known`.
+  ///
+  /// The result of this call is stored in [wellKnown] for later use at runtime.
+  @override
+  Future<DiscoveryInformation> getWellknown([String? hostname]) async {
+    final wellKnown = await getDiscoveryInformationsByUserId(
+      hostname ?? baseUri!.host,
```

Comment:
This could be a problem. Imagine you have a homeserver example.com delegating to matrix.example.com. On the first call this would request example.com/.well-known/matrix/client, on the second call it would request <b>matrix.</b>example.com/.well-known/matrix/client. Those may contain different values or the latter might not exist at all. I think we should always request the one from the userids servername unless we are pre-login.